Month3/Project1/Project1/ArrayVectors.cpp

```diff
@@ -0,0 +1,280 @@
+#include "ArrayVectors.h"
+#include "TStaticArray.h"
+#include "Circle.h"
+#include "Square.h"
+#include "TDinamicArray.h"
+
+
+
+FShape::FShape()
+{
+
+}
+
+FCircle::FCircle(float& NewRadius)
+{
+	Radius = NewRadius;
+
+}
+FSquare::FSquare(float& NewSideLenght) 
+{
+	SideLenght = NewSideLenght;
+}
+
+
+
+int ValidateIntInput()
+{
+	int Input;
+	std::cin.ignore(100, '\n');
+	while (std::cin.peek() == '\n')
+	{
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid input, try again" << std::endl;
+	}
+
+	std::cin >> Input;
+	while (!std::cin.good() || Input <= 0)
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid input,try again" << std::endl;
+		std::cin >> Input;
+	}
+	return Input;
+}
+float ValidateFloatInput()
+{
+	float Input;
+	std::cin.ignore(100, '\n');
+
+	while (std::cin.peek() == '\n')
+	{
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid input, try again" << std::endl;
+	}
+	std::cin >> Input;
+	while (!std::cin.good() || Input <= 0)
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid input,try again" << std::endl;
+		std::cin >> Input;
+	}
+	return Input;
+}
+
+
+
+
+
+float FCircle::GetPerimeter() const
+{
+	return (float)M_PI * (float)pow(Radius, 2);
+}
+
+float FCircle::GetArea() const
+{
+	const float Area = (2 * M_PI) * Radius;
+	return  Area;
+}
+
+
+
+
+float FSquare::GetArea() const
+{
+	return (float)pow(SideLenght, 2);
+}
+float FSquare::GetPerimeter() const
+{
+	return SideLenght * 4;
+}
+
+
+
+
+void BuiltInArray()
+{
+	std::cout << "How many shapes you want to create" << std::endl;
+	int ShapesNumber = ValidateIntInput();
+	FShape** ArrShapes = new FShape * [ShapesNumber];
+	int Position;
```

Comment:
Remember that you should only declare your variables where you need them. If you need a variable position inside a for-loop, why do you declare it all the way up here, outside of the loop? Same thing with the rest of these variables.